File: src/main/java/ru/gb/university/services/StudentService.java
package ru.gb.university.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import ru.gb.university.model.Student;
import ru.gb.university.repositories.StudentRepository;

import java.util.Collections;
import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class StudentService {
    private final StudentRepository studentRepository;

    public List<Student> findAll() {
        return Collections.unmodifiableList(studentRepository.findAll());
    }

    public void add(String name, int age) {
        studentRepository.save(new Student(name, age));
    }

    public Optional<Student> get(Long id) {
        return studentRepository.findById(id);
    }

    @Transactional
    public void delete(Long id) {
        studentRepository.deleteById(id);
    }

    public void updateName(Long id, String name) {
        studentRepository.updateName(id, name);
    }

    public void updateAge(Long id, int age) {
        studentRepository.updateAge(id, age);
    }
}

File: frontend_university/src/app/service/student.service.ts
import { Injectable } from '@angular/core';
import {Student} from "./student";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class StudentService {

  constructor(public http: HttpClient) { }

  public findAll(){
    return this.http.get<Student[]>("/university/students");
  }

  public findByID(id: number){
    return this.http.get<Student>(`/university/students/${id}/info`);
  }

  public delete(id: number){
    return this.http.delete(`/university/students/${id}`);
  }

  public updateName(id: number, name: string){
    return this.http.put(`/university/students/name/${id}`, name);
  }

  public updateAge(id: number, age: number){
    return this.http.put(`/university/students/age/${id}`, age);
  }

  public save(student: Student){
      return this.http.post(`/university/students`, student);
  }
}
